src/ruuvitag_ble/parser.py

```diff
@@ -1,16 +1,26 @@
 from __future__ import annotations
 
 import logging
+from typing import Type, Union
 
 from bluetooth_data_tools import short_address
 from bluetooth_sensor_state_data import BluetoothData
 from home_assistant_bluetooth import BluetoothServiceInfo
 from sensor_state_data import DeviceClass, Units
 
+from ruuvitag_ble.df3_decoder import DataFormat3Decoder
 from ruuvitag_ble.df5_decoder import DataFormat5Decoder
 
 _LOGGER = logging.getLogger(__name__)
 
+decoder_classes: dict[
+    int,
+    Type[Union[DataFormat3Decoder, DataFormat5Decoder]],
+] = {
```

Comment:
As noted above, futuristic types are fine:
```suggestion
decoder_classes: dict[
    int,
    type[DataFormat3Decoder | DataFormat5Decoder],
] = {
```